LibJS: Implement resizable ArrayBuffers and support for them in TypedArray/DataView

PrestonLTaylor opened the pull request:
This pr supports resizable ArrayBuffers, as well as support for TypedArray/DataViews to have resizable ArrayBuffers views.

Test262 Diff: ✅ +240 ❌ -240 :^)

Closes #21048

trflynn89 commented:
Did you test this locally? `test-js` is failing quite hard here.

FWIW I was implementing this as well: https://github.com/trflynn89/serenity/tree/resizable_array_buffer

While doing so, I ran into spec bugs that will prevent us from implementing these changes. So far:
`https://github.com/tc39/ecma262/issues/3199`

PrestonLTaylor commented:
> Did you test this locally? `test-js` is failing quite hard here.
> 
> FWIW I was implementing this as well: https://github.com/trflynn89/serenity/tree/resizable_array_buffer
> 
> While doing so, I ran into spec bugs that will prevent us from implementing these changes. So far: `https://github.com/tc39/ecma262/issues/3199`

Woops, I completely forgot to run test-js. I re-ran them locally and fixed the issues.

For the spec bug, the author of the proposal has made a pr to fix the bug and I've implemented them in `RevalidateAtomicAccess`.
